src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -398,13 +398,27 @@ private ComponentState GetOptionalComponentState(int componentId)
                 ? componentState
                 : null;
 
+        /// <summary>
+        /// Processses pending renders requests from components if there are any.
+        /// </summary>
+        protected virtual void ProcessPendingRender()
+        {
+            ProcessRenderQueue();
+        }
+
         private void ProcessRenderQueue()
         {
+            EnsureSynchronizationContext();
```

Comment:
This can go into `ProcessPendingRender`, since that's the external entrypoint, right?

Reply:
I thought it couldn't because it was on the virtual method, but not that I think again it can, as people is forced to invoke the base to do any meaningful work.

Reply:
I'll change this in a separate PR as I don't want to re-run the build